text/0000-futures.md

```diff
@@ -154,31 +164,316 @@ can be dropped.
 
 ### Waking up
 
-Each task executor provides its own scheduling facilities, and hence needs to
-customize the way task wakeups are handled. Most of the time, you should use the
-`std::task::Wake` trait defining wakeup behavior:
+If a future can not be directly fulfilled during execution and returns `Pending`,
+it needs a way to later on inform the executor that it needs to get polled again
+to make progress.
+
+This functionality is provided through a set of `Waker` types.
+
+`Waker`s are objects which are passed as a parameter to the `Future::poll` call,
+and which can be stored by the implementation of those `Futures`s. Whenever a
+`Future` has the need to get polled again, it can use the `wake` method of the
+waker in order to inform the executor that the task which owns the `Future`
```

Comment:
```suggestion
waker to inform the executor that the task which owns the `Future`
```

Reply:
I actually find it clearer that way, but that might be my non native speaker understanding. Open for further feedback around this (and the similar open things).

Thanks for all the suggestions!

Reply:
Generally speaking, the phrase "in order to" can be replaced with "to" since the former is *redundant*. It also contributes to the length and *wordiness* of a sentence which is good to avoid because you only have so much *"attention budget"* when people are reading.

- https://blog.apastyle.org/apastyle/2015/09/principles-of-writing-how-to-avoid-wordiness.html
- https://www.oxford-royale.co.uk/articles/clumsy-phrases-avoid.html

I'm not a native English speaker myself, but this advice is what my English-technical-writing professors gave ;)